addons/api/tests/unit/serializers/managed-groups-test.js

```diff
@@ -0,0 +1,127 @@
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+
+module('Unit | Serializer | Managed groups', function (hooks) {
+  setupTest(hooks);
+
+  test('it serializes correctly on create with no filter provided', function (assert) {
+    assert.expect(1);
+    const store = this.owner.lookup('service:store');
+    const serializer = store.serializerFor('managed-groups');
+    const record = store.createRecord('managed-groups', {
+      name: 'Group',
+      description: 'Description',
+      member_ids: ['1', '2'],
+      version: 1,
+      auth_method_id: '1234',
+    });
+    const snapshot = record._createSnapshot();
+    snapshot.adapterOptions = {};
+    const serializedRecord = serializer.serialize(snapshot);
+    assert.deepEqual(serializedRecord, {
+      name: 'Group',
+      description: 'Description',
+      version: 1,
+      auth_method_id: '1234',
+      type: 'OIDC',
+      attributes: {
+        filter: null,
+      },
+    });
+  });
+
+  test('it serializes correctly on create when filter provided', function (assert) {
+    assert.expect(1);
+    const store = this.owner.lookup('service:store');
+    const serializer = store.serializerFor('managed-groups');
+    const record = store.createRecord('managed-groups', {
+      name: 'Group',
+      description: 'Description',
+      member_ids: ['1', '2'],
+      version: 1,
+      auth_method_id: '1234',
+      type: 'OIDC',
+      attributes: {
+        filter: 'key=value',
+      },
+    });
+    const snapshot = record._createSnapshot();
+    snapshot.adapterOptions = {};
+    const serializedRecord = serializer.serialize(snapshot);
+    assert.deepEqual(serializedRecord, {
+      name: 'Group',
+      description: 'Description',
+      version: 1,
+      auth_method_id: '1234',
+      type: 'OIDC',
+      attributes: {
+        filter: 'key=value',
+      },
+    });
+  });
+
+  test('it does not serialize readOnly attributes', function (assert) {
```

Comment:
Even tho other creation tests are validating this. This test is VERY specific passing the three `readOnly` attributes the model has and making sure we did not serialize anything there.